relay/api/resources.py

```diff
@@ -431,18 +432,35 @@ def post(self, args, network_address: str):
         value = args["value"]
         max_fees = args["maxFees"]
         max_hops = args["maxHops"]
+        sender_pays_fees = args["sender_pays_fees"]
 
-        cost, path = self.trustlines.currency_network_graphs[network_address].find_path(
-            source=source,
-            target=target,
-            value=value,
-            max_fees=max_fees,
-            max_hops=max_hops,
-            timestamp=timestamp,
-        )
+        if sender_pays_fees:
+            cost, path = self.trustlines.currency_network_graphs[
+                network_address
+            ].find_transfer_path_sender_pays_fees(
+                source=source,
+                target=target,
+                value=value,
+                max_fees=max_fees,
+                max_hops=max_hops,
+                timestamp=timestamp,
+            )
+        else:
+            cost, path = self.trustlines.currency_network_graphs[
+                network_address
+            ].find_transfer_path_receiver_pays_fees(
+                source=source,
+                target=target,
+                value=value,
+                max_fees=max_fees,
+                max_hops=max_hops,
+                timestamp=timestamp,
+            )
 
         return _estimate_gas_for_transfer(
-            self.trustlines, PaymentPath(cost, path, value), network_address
+            self.trustlines,
+            PaymentPath(cost, path, value, sender_pays_fees=sender_pays_fees),
```

Comment:
What is your reason to add `sender_pays_fees` here? If you want to return in from the api, you are missing the schema.

Reply:
Because to estimate the gas necessary for the transaction, you need to know whether you estimate the gas for the function `transfer` or `transferReceiverPays` on the currency network contract (https://github.com/trustlines-network/relay/pull/280/files#diff-fb784d1467ed2b14b3b247b1e09307e3R226).
Or do I misunderstand you?

I do not want to return this value.

Reply:
Why do you not want to return that value? I think for clarity it would be good to do so.

Reply:
IYou also need to change the function here https://github.com/trustlines-network/relay/blob/bcaf98519ac599508087bf39fca1a391fe1b816c/relay/api/resources.py#L406. 
It will return the Payment path but always with sender as the payer, which mught be wrong. 
I would say this is partially because the name of this function is bad. We need to give it a better name that describe what it is actually doing: estimating the gas and then returning a path object according to that information.